qiling/extensions/idaplugin/qilingida.py

```diff
@@ -2059,11 +2061,16 @@ def ql_set_rootfs(self):
 
         rootfspath = setupdlg.path_name.value
         customscript = setupdlg.script_name.value
+        parameter = setupdlg.parameters.value
         env = setupdlg.env_var.value
 
         if customscript != '':
             self.customscriptpath = customscript
 
+        para_array = parameter.split(' ')
```

Comment:
Need a more robust parser. e.g. parsing strings including spaces.

Reply:
Probably a regex or a proper grammar for ast could do the job, WDYT?

Reply:
Or we can just demand everything from the user, and ask for a JSON as we did for the custom ENV.

Reply:
Not sure. One case I'm considering is like:

`printf "Hello %s" "world!"`

Reply:
A relevant post: https://stackoverflow.com/questions/1706551/parse-string-into-argv-argc

Reply:
Just have a user-supplied JSON (from a file)?

e.x:
{"-c","aaaaa","bbbbb","\"c deefedede\""}

Reply:
> A relevant post: https://stackoverflow.com/questions/1706551/parse-string-into-argv-argc

BTW, I will look at it, in the next few days.

Reply:
I'm now using `shlex` in order to parse arguments @wtdcode.
Input: `-c aaaa -j "aa bb"`

parsed: `['-c', 'aaaa', '-j', 'aa bb']`